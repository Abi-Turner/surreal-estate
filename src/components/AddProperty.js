import React, { useState } from "react";
import "../styles/add-property.css";

const AddProperty = () => {
  const initialState = {
    fields: {
      title: "",
      city: "Manchester",
    },
  };

  const [fields, setFields] = useState(initialState.fields);

  const handleAddProperty = (event) => {
    event.preventDefault();
    // eslint-disable-next-line no-console
    console.log(fields);
  };

  const handleFieldChange = (event) => {
    setFields({ ...fields, [event.target.name]: event.target.value });
  };

  return (
    <div className="add-property">
      Add Property Page
      <form onSubmit={handleAddProperty}>
        <div className="form-field">
          <label htmlFor="title">
            Title
            <input
              id="title"
              name="title"
              value={fields.title}
              onChange={handleFieldChange}
            />
          </label>
        </div>
        <div className="form-field">
          <label htmlFor="city">
            City
            <select
              id="city"
              name="city"
              value={fields.city}
              onChange={handleFieldChange}
            >
              <option value="Manchester">Manchester</option>
              <option value="Leeds">Leeds</option>
              <option value="Sheffield">Sheffield</option>
              <option value="Liverpool">Liverpool</option>
            </select>
          </label>
        </div>
        <div className="form-field">
          <label htmlFor="type">
            Type
            <select
              id="type"
              name="type"
              value={fields.type}
              onChange={handleFieldChange}
            >
              <option value="Flat">Flat</option>
              <option value="Detached">Detached</option>
              <option value="Semi-Detached">Semi-Detached</option>
              <option value="Terraced">Terraced</option>
              <option value="End of Terrace">End of Terrace</option>
              <option value="Cottage">Cottage</option>
              <option value="Bungalow">Bungalow</option>
            </select>
          </label>
        </div>
        <div className="form-field">
          <label htmlFor="bedrooms">
            Bedrooms
            <input
              id="bedrooms"
              name="bedrooms"
              value={fields.bedrooms}
              onChange={handleFieldChange}
            />
          </label>
        </div>
        <div className="form-field">
          <label htmlFor="bathrooms">
            Bathrooms
            <input
              id="bathrooms"
              name="bathrooms"
              value={fields.bathrooms}
              onChange={handleFieldChange}
            />
          </label>
        </div>
        <div className="form-field">
          <label htmlFor="price">
            Price{" "}
            <input
              id="price"
              name="price"
              value={fields.price}
              onChange={handleFieldChange}
            />
          </label>
        </div>
        <div className="form-field">
          <label htmlFor="email">
            Email
            <input
              name="email"
              value={fields.email}
              onChange={handleFieldChange}
            />
          </label>
        </div>
        <button type="submit">Add</button>
      </form>
    </div>
  );
};

export default AddProperty;
