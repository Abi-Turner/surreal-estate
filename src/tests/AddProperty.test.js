import { render } from "@testing-library/react";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import AddProperty from "../components/AddProperty";

describe("App", () => {
  const { asFragment } = render(
    <BrowserRouter basename="/">
      <AddProperty />
    </BrowserRouter>
  );

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});