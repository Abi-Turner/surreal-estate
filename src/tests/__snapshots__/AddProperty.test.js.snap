// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="add-property"
  >
    Add Property Page
    <form>
      <div
        class="form-field"
      >
        <label
          for="title"
        >
          Title
          <input
            id="title"
            name="title"
            value=""
          />
        </label>
      </div>
      <div
        class="form-field"
      >
        <label
          for="city"
        >
          City
          <select
            id="city"
            name="city"
          >
            <option
              value="Manchester"
            >
              Manchester
            </option>
            <option
              value="Leeds"
            >
              Leeds
            </option>
            <option
              value="Sheffield"
            >
              Sheffield
            </option>
            <option
              value="Liverpool"
            >
              Liverpool
            </option>
          </select>
        </label>
      </div>
      <div
        class="form-field"
      >
        <label
          for="type"
        >
          Type
          <select
            id="type"
            name="type"
          >
            <option
              value="Flat"
            >
              Flat
            </option>
            <option
              value="Detached"
            >
              Detached
            </option>
            <option
              value="Semi-Detached"
            >
              Semi-Detached
            </option>
            <option
              value="Terraced"
            >
              Terraced
            </option>
            <option
              value="End of Terrace"
            >
              End of Terrace
            </option>
            <option
              value="Cottage"
            >
              Cottage
            </option>
            <option
              value="Bungalow"
            >
              Bungalow
            </option>
          </select>
        </label>
      </div>
      <div
        class="form-field"
      >
        <label
          for="bedrooms"
        >
          Bedrooms
          <input
            id="bedrooms"
            name="bedrooms"
            value=""
          />
        </label>
      </div>
      <div
        class="form-field"
      >
        <label
          for="bathrooms"
        >
          Bathrooms
          <input
            id="bathrooms"
            name="bathrooms"
            value=""
          />
        </label>
      </div>
      <div
        class="form-field"
      >
        <label
          for="price"
        >
          Price 
          <input
            id="price"
            name="price"
            value=""
          />
        </label>
      </div>
      <div
        class="form-field"
      >
        <label
          for="email"
        >
          Email
          <input
            name="email"
            value=""
          />
        </label>
      </div>
      <button
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
